pydantic/type_adapter.py

```diff
@@ -120,40 +121,13 @@ class TypeAdapter(Generic[T]):
         serializer: The schema serializer for the type.
     """
 
-    if TYPE_CHECKING:
-
-        @overload
-        def __new__(cls, __type: type[T], *, config: ConfigDict | None = ...) -> TypeAdapter[T]:
-            ...
-
-        # this overload is for non-type things like Union[int, str]
-        # Pyright currently handles this "correctly", but MyPy understands this as TypeAdapter[object]
-        # so an explicit type cast is needed
-        @overload
-        def __new__(cls, __type: T, *, config: ConfigDict | None = ...) -> TypeAdapter[T]:
-            ...
-
-        def __new__(cls, __type: Any, *, config: ConfigDict | None = None) -> TypeAdapter[T]:
-            """A class representing the type adapter."""
-            raise NotImplementedError
-
-        @overload
-        def __init__(
-            self, type: type[T], *, config: ConfigDict | None = None, _parent_depth: int = 2, module: str | None = None
-        ) -> None:
-            ...
-
-        # this overload is for non-type things like Union[int, str]
-        # Pyright currently handles this "correctly", but MyPy understands this as TypeAdapter[object]
-        # so an explicit type cast is needed
-        @overload
-        def __init__(
-            self, type: T, *, config: ConfigDict | None = None, _parent_depth: int = 2, module: str | None = None
-        ) -> None:
-            ...
-
     def __init__(
-        self, type: Any, *, config: ConfigDict | None = None, _parent_depth: int = 2, module: str | None = None
+        self,
+        type: type[T],
```

Comment:
Does this work now? What about passing in `int | str`?

Reply:
Following up, this doesn't work:

```py
from pydantic import TypeAdapter

from typing import Union

adapter: TypeAdapter[Union[int, str]] = TypeAdapter(Union[int, str])
```
--> 
```
test.py:5: error: Argument 1 to "TypeAdapter" has incompatible type "object"; expected "Union[Type[int], Type[str]]"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

Reply:
@Viicos,

Is there any downside of leaving the above overloads within the `if TYPE_CHECKING` block? I think it'd be preferred to leave those in place.